br/pkg/backup/prepare_snap/prepare.go

```diff
@@ -414,17 +422,31 @@ func (p *Preparer) pushWaitApply(reqs pendingRequests, region Region) {
 	p.inflightReqs[region.GetMeta().Id] = *region.GetMeta()
 }
 
-func (p *Preparer) prepareConnections(ctx context.Context) error {
+// PrepareConnections prepares the connections for each store.
+// This will pause the admin commands for each store.
+func (p *Preparer) PrepareConnections(ctx context.Context) error {
 	log.Info("Preparing connections to stores.")
 	stores, err := p.env.GetAllLiveStores(ctx)
 	if err != nil {
 		return errors.Annotate(err, "failed to get all live stores")
 	}
+
+	log.Info("Start to initialize the connections.", zap.Int("stores", len(stores)))
+	clients := map[uint64]PrepareClient{}
 	for _, store := range stores {
-		_, err := p.streamOf(ctx, store.Id)
+		cli, err := p.env.ConnectToStore(ctx, store.Id)
 		if err != nil {
-			return errors.Annotatef(err, "failed to prepare connection to store %d", store.Id)
+			return errors.Annotatef(err, "failed to dial the store %d", store.Id)
+		}
+		clients[store.Id] = cli
+	}
+
+	for id, cli := range clients {
```

Comment:
why use two loops? seems it can be done without `clients` map

Reply:
The first `createAndCacheStream` must happens after ALL connections established. I think we cannot remove this. Or would you give a code snippet?

Reply:
for _, store := range stores {
 		cli, err := p.env.ConnectToStore(ctx, store.Id)
 		if err != nil {
 			return errors.Annotatef(err, "failed to dial the store %d", store.Id)
 		}
        if err := p.createAndCacheStream(ctx, cli, store.Id); err != nil {
 			return errors.Annotatef(err, "failed to create and cache stream for store %d", id)
 		}	
 	}

Reply:
THis will make `createAndCacheStream` concurrency with `ConnectToStore`, recall that:
> The first createAndCacheStream must happens after ALL connections established.

Reply:
...then what's problem with that?